subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/DefaultExternalResourceArtifactResolver.java

```diff
@@ -117,6 +117,16 @@ private LocallyAvailableExternalResource downloadByUrl(List<ResourcePattern> pat
         return null;
     }
 
+    private String normalizeRelativeUrl(UrlBackedArtifactMetadata artifact) {
+        ModuleComponentIdentifier componentId = artifact.getComponentId();
+        if (componentId instanceof MavenUniqueSnapshotComponentIdentifier) {
+            // We need to replace the `-SNAPSHOT` in the relative URL but only for the version part
+            MavenUniqueSnapshotComponentIdentifier snapshotComponentId = (MavenUniqueSnapshotComponentIdentifier) componentId;
+            return artifact.getRelativeUrl().replace("-" + snapshotComponentId.getSnapshotVersion(), "-" + snapshotComponentId.getTimestampedVersion());
```

Comment:
Corner case: can the snapshot version appear anywhere else in the URL where we don't actually need to replace it? Do we want to replace only the specific occurrences? Or is this not a realistic case?

Reply:
Yes it does, as a `/<version>/` path element, which is why the replacement is done for `-<version>` which appears only in the file name according to Maven repository conventions.